<!-- 
Files tuned for ethzasl_icp_mapping commit 52d78c96c2cf2f58a400c649c561dbfec95e41a4
and for libpointmatcher commit e048c4d1180b1de005af072bf137ea4d8cf2b826

"Long-term 3D map maintenance in dynamic environments"
-->

<launch>
	<!--
	<arg name="file_path" default="/home/fabio/Documents/ASL-Lidar/data/velodyne/2016-03-11/2016-03-11-12-11-53.bag"/>
	-->
	

	
	<!-- <arg name="rviz_config" /> -->
	
	<!--
	<node pkg="rviz" type="rviz" name="visualization" output="screen" respawn="true" args="-d /home/fabio/Documents/ASL-Lidar/data/velodyne/rvis/velodyne16.rviz" />
	-->
<!--
    <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 odom velodyne 100" />
-->

	<node name="mapper" type="dynamic_mapper" pkg="ethzasl_icp_mapper" output="screen" >
		
		<param name="subscribe_scan" value="false"/>
		<param name="subscribe_cloud" value="true"/>

		<remap from="cloud_in" to="/fredeluga/velodyne_points"/>
		
		<param name="icpConfig" value="$(find ethzasl_icp_mapper)/cfg/icp_dynamic.yaml" />
		<param name="inputFiltersConfig" value="$(find ethzasl_icp_mapper)/cfg/input_filters.yaml" />
		<param name="mapPostFiltersConfig" value="$(find ethzasl_icp_mapper)/cfg/map_post_filters.yaml" />
		

		<param name="odom_frame" value="/velodyne"/>
		<param name="map_frame" value="/map" />

		<param name="useROSLogger" value="false" />
		<param name="inputQueueSize" value="1" />
		<param name="tfRefreshPeriod" value="0.005" />
		<param name="minOverlap" value="0.2" /> 
		<param name="maxOverlapToMerge" value="0.99" /> 
		<param name="minReadingPointCount" value="1000" /> 
		<param name="minMapPointCount" value="10000" />
		<param name="localizing" value="true" /> 
		<param name="mapping" value="true" />

		<param name="priorStatic" value="0.45"/>
		<param name="priorDyn" value="0.55"/>
		<param name="maxAngle" value="0.01"/>
		<param name="eps_a" value="0.1"/> 
		<param name="eps_d" value="0.1"/>
		<param name="alpha" value="0.99"/>
		<param name="beta" value="0.90"/>
		<param name="maxDyn" value="0.90"/>
		<param name="maxDistNewPoint" value="0.7"/>
		<param name="sensorMaxRange" value="60.0"/>
	</node>

</launch>
